### DISCUSSION TITLE
Cannot find image in PDF page where image exist

### DISCUSSION THREAD

**Discussion by dannydocrxiv** [pymupdf/PyMuPDF#1470] [Q&A] [answered]
Please check attached PDF on page 13 out of 25
The top part is an image but page.get_images() / page.get_image_info() does not return any image on this page
I don't get any text for it also 
The block types are all zero for this page also (using  page.get_text('dict'))
[PeerJ_2021_May_28_d5de.pdf](https://github.com/pymupdf/PyMuPDF/files/7737945/PeerJ_2021_May_28_d5de.pdf)

**Comment by JorjMcKie** [comment]
Well, you are wrong:
It is no image, but a (complex) set of drawings.

**Comment by dannydocrxiv** [comment]
ok. 
Any idea how to extract the text from this complex drawings

**Comment by dannydocrxiv** [comment]
Or at least know I have this case and do OCR on the full page in these cases

**Comment by JorjMcKie** [comment, marked as answer]
> Or at least know I have this case and do OCR on the full page in these cases

This page has a large amount of drawings, which could raise initial suspicions to start with:
```python
>>> doc=fitz.open("PeerJ_2021_May_28_d5de.pdf")
>>> page=doc[12]
>>> paths = page.get_drawings()
>>> len(paths)
502
>>>
```

If you then look at the amount of single draw commands:
```python
>>> draws = 0
>>> for p in paths:
        draws += len(p["items"])
	
>>> draws
4623
>>>
```
... it becomes clear that something significant is being done on the page.

I have a script somewhere which helps separating drawings into disjoint subsets. Once you have those, join associated rectangles and make pixmaps, which can be separately be OCRed.

**Reply by dannydocrxiv** [reply]
Ok, Thx
